packages/node_modules/@ciscospark/redux-module-conversation/src/reducer.js

```diff
@@ -65,17 +58,18 @@ export default function reducer(state = initialState, action) {
     let activities = new OrderedMap(conversationActivities.map((activity) => [activity.url, activity]));
     activities = activities.sortBy((activity) => activity.published);
 
-    return Object.assign({}, state, {
-      activities,
-      defaultActivityEncryptionKeyUrl,
-      id,
-      kmsResourceObjectUrl,
-      locusUrl,
-      url,
-      isFetching: false,
-      isLoaded: true,
-      participants: participants.items
-    });
+    return state
+      .set(`activities`, activities)
+      .merge({
+        defaultActivityEncryptionKeyUrl,
```

Comment:
I like the idea of not converting before sending. We could create a utility method that gets the items the sdk needs out of the immutable object and just have it call that method right before the sdk. That way we don't have to `.toJS` the entire conversation object, but I don't know how much effort that would be and time saved.